src/jshint.js

```diff
@@ -4337,31 +4357,41 @@ var JSHINT = (function() {
           break;
         }
       }
+      if (state.tokens.next.value === "from") {
```

Comment:
In a follow-up diff, the linting behavior of *export* should be implemented more thoroughly.

The body of the *ExportClause* needs to be linted differently depending on this piece of information.

With "undef" and "unused" turned on:

Should pass:
```js
var foo = "foo";
export { foo };
```

Should fail "undef" (currently does not)
```js
export { foo };
```

Should pass (currently does, but because above is broken)
```js
export { foo } from "source";
```

Should fail "unused" (currently does not)
```js
var foo = "foo";
export { foo } from "source";
```